### PR TITLE
New version of propose_changelog.py

### PR DESCRIPTION
**Pull request by chillenzer**
As requested by @psychocoderHPC, here's a quick rewrite of `propose_changelog.py`. One matter of taste might be that I have chosen to depend on `pyyaml` for the pretty printing of nested dictionaries. That is technically not necessary but I really don't care for writing all the formatting by hand. This comes with the tiny drawback that all the strings containing  `#` are surrounded by single quotes. Theoretically, the formatting can be adjusted arbitrarily by writing custom dumpers but then again, we would be back at writing custom formatting code.

<details><summary>Output at the time of writing</summary>

### Output

```
Bug Fixes:
  PIC:
  - 'fix Gaussian and Dispersive pulse #4889'
  PMacc:
  - 'fix `atomicAllInc()` #4825'
  - 'avoid using uninitialized `pmacc::memory::Array` #4789'
  other:
  - 'Correct spack options for boost #4780'
  - 'fix potential deadlock #4755'
  plugins: []
  tools:
  - 'fix slow tbg substitution #4905'
  - 'fix growth rate and charge conservation test tools #4834'
Documentation:
  PIC:
  - 'replace cupla fully by alpaka #4838'
  PMacc:
  - 'replace cupla fully by alpaka #4838'
  other:
  - 'Correct spack options for boost #4780'
  - 'RISC-V support #4712'
  plugins:
  - 'Add shadowgraphy plugin #4868'
  - 'remove deactivation of ISAAC with HIP #4821'
  - 'change class and function names in radPlugin to camelCase #4734'
  - 'add calorimeter python module #4697'
  tools:
  - 'Pre-commit #4792'
  - 'Remove hemera k80 and k20 partition #4779'
  - 'add calorimeter python module #4697'
Features:
  PIC:
  - 'Add feature `--dump-metadata` #4812'
  - 'PIConGPU unit tests #4723'
  - 'add CKC Solver #4661'
  PMacc:
  - 'Add PMacc example - Solving the 2D HeatEquation #4767'
  other:
  - 'RISC-V support #4712'
  - 'Make fieldAbsorberTest a test for the CI #4682'
  - 'Add CurrentDeposition test #4557'
  plugins:
  - 'Add shadowgraphy plugin #4868'
  - 'remove deactivation of ISAAC with HIP #4821'
  - 'Implement checkpoint and restart capability to the binning plugin #4766'
  - 'change class and function names in radPlugin to camelCase #4734'
  - 'add calorimeter python module #4697'
  - 'change radiation plugin python module to openPMD-api #4680'
  tools:
  - 'Remove hemera k80 and k20 partition #4779'
  - 'add Snakemake workflow example #4774'
  - 'add calorimeter python module #4697'
  - 'change radiation plugin python module to openPMD-api #4680'
Refactoring:
  PIC:
  - 'replace cupla fully by alpaka #4838'
  - 'optimize collision #4828'
  - 'move PMacc and PIConGPU fully to alpaka #4820'
  - 'Topic remove preprocessor struct generator #4750'
  - 'refactor to make superCellSize and frameSize independant #4685'
  - 'refactor particle init kernel #4681'
  - 'Add an optional time delay for incident field pulses #4651'
  PMacc:
  - 'rename `getDataSpace` to `capacityND` #4851'
  - 'refactor lockstep programming model #4840'
  - 'replace cupla fully by alpaka #4838'
  - 'move PMacc and PIConGPU fully to alpaka #4820'
  - 'Topic remove preprocessor struct generator #4750'
  - 'remove accessor from `math::Vector<>` #4746'
  - 'Get supercell index relative to user defined origin #4731'
  - 'remove `atomicAddNoRet()` #4722'
  - 'refactor DataSpaceOperations #4719'
  other:
  - 'remove CONST_VECTOR from OnePositionParameter #4875'
  plugins:
  - 'optimize OpenPMD IO #4882'
  - 'openPMD: add parameter `particleIOChunkSize` #4874'
  - 'Bump openPMD-api to minimal version 0.15 #4782'
  - 'simplify output structure colorimeter plugin  #4695'
  tools:
  - 'Pre-commit #4792'
User Input Changes:
- 'remove CONST_VECTOR from OnePositionParameter #4875'
- 'Topic remove preprocessor struct generator #4750'
- 'change class and function names in radPlugin to camelCase #4734'
- 'refactor to make superCellSize and frameSize independant #4685'
```
</details>

### CODE REVIEW

File: docs/propose_changelog.py
Comment (PrometheusPi) [inline docs/propose_changelog.py]: why `GH_PTA` not `GH_PAT`?

File: docs/propose_changelog.py
Comment (PrometheusPi) [inline docs/propose_changelog.py:41]: 🤣

Comment (PrometheusPi) [reply docs/propose_changelog.py:41]: Where is my [bottle of wine](https://www.bbc.com/news/articles/c84z2jqpvpko)?

File: docs/propose_changelog.py
Comment (PrometheusPi) [inline docs/propose_changelog.py]: What is `func`?

File: docs/propose_changelog.py
Comment (PrometheusPi) [inline docs/propose_changelog.py:115]: You like C/C++, right?

File: docs/propose_changelog.py
Comment (PrometheusPi) [inline docs/propose_changelog.py]: I think a help would be useful, possible via [`argparse`](https://docs.python.org/3/library/argparse.html)

**Review by PrometheusPi — CHANGES_REQUESTED** [review]
Looks good, just a help (as used by all other tools) would be a nice to have.

File: docs/propose_changelog.py
Comment (chillenzer) [reply docs/propose_changelog.py:115]: I agree that it is mostly habitual in this specific case but not necessarily from C/C++: https://realpython.com/python-main-function/#create-a-function-called-main-to-contain-the-code-you-want-to-run

File: docs/propose_changelog.py
Comment (chillenzer) [reply docs/propose_changelog.py]: Yeah, I know. But I dislike `argparse` and  prefer [click](https://click.palletsprojects.com). But then again, that's an (almost) unnecessary (non-standard-library) dependency, so for scripts as simple as this one, I typically go with just `sys.argv` directly until a cautious reviewer forces me to use one or the other. Consider it done.

File: docs/propose_changelog.py
Comment (chillenzer) [reply docs/propose_changelog.py:41]: That is the third possible reaction to this, which I didn't see coming:
1. Don't read it and don't request changes.
2. Read it and request changes.
4. Read it but request something completely different.

File: docs/propose_changelog.py
Comment (chillenzer) [reply docs/propose_changelog.py:115]: Even more "official": https://docs.python.org/3/library/__main__.html#idiomatic-usage

**Review by PrometheusPi — DISMISSED** [review]

**Comment by chillenzer** [comment]
As the CI does something weird currently and @psychocoderHPC will most likely be the main user of this, we'll wait for your approval for this.

**Comment by psychocoderHPC** [comment]
@chillenzer Can we automatically run a short regex over the output to remove `'` if we have it as last and first letter?

**Comment by chillenzer** [comment]
Hi, I've gone the simple route of just removing all quotes. I would argue that the changelog will experience minor human adjustment anyways, so adding a single pair of missing quotes every few releases won't hurt that much.

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**
